src/System.Private.ServiceModel/src/System/ServiceModel/Channels/TextMessageEncoder.cs

```diff
@@ -546,7 +546,9 @@ public override Task<ArraySegment<byte>> WriteMessageAsync(Message message, int
                     await message.WriteMessageAsync(xmlWriter);
                     xmlWriter.WriteEndDocument();
                 }
-                await xmlWriter.FlushAsync();
+
+                // The underlying type of xmlWriter, XmlUTF8TextWriter, has not implemented FlushAsync() yet.
+                xmlWriter.Flush();
```

Comment:
@roncain, I would prefer to keep the call to xmlWriter.FlushAsync and let the xmlWriter decide what to do. If the xmlWriter is doing nothing right now, then that's a bug that needs to be fixed. But the fix could simply be for xmlWriter to either directly call the synchronous Flush method or to call Task.Run(). xmlWriter is the correct place to decide what to do as it is likely easy to check if there's anything to flush and if there isn't, return, otherwise call Task.Run to invoke Flush(). This wouldn't require converting everything to async and would allow a fast path for the no-op situation avoiding starting a new task on another thread just to no-op on that thread and mark the Task complete.